doc/content/integrations/payload-formatters/javascript/uplink/_index.md

```diff
@@ -0,0 +1,256 @@
+
+```js
+{
+  "time": "2022-08-29T13:50:15Z", // Timestamp (RFC 3339)
+  "air": {
+    "pressure": 1032,         // Atmospheric pressure (hPa): [900..1100]
+    "relativeHumidity": 43.9, // Relative humidity (%): [0..100]
+    "temperature": 21.5       // Temperature (Celcius): [-273.15..)
+  },
+  "wind": {
+    "direction": 321, // Direction (degrees): [0..360)
+    "speed": 5.2      // Speed (m/s): [0..)
+  }
+}
+```
+
+### Advanced: Return Multiple Measurements {#multiple-measurements}
+
+Multiple measurements can be encoded by the end device in a single LoRaWAN uplink message. This is a good practice as it is more battery efficient. Therefore, the uplink decoder may find multiple measurements from the same sensor. The normalizer function can therefore return an array of measurements, as an alternative to a single measurement:
+
+```js
+function normalizeUplink(input) {
+  // input has the following structure:
+  // {
+  //   "data": {
+  //     "readings": [
+  //       { "temperature": 51.4 },
+  //       { "temperature": 50.1 },
+  //       { "temperature": 56.8 }
+  //     ]
+  //   }
+  // }
+  var data = [];
+  for (var i = 0; i < input.data.readings.length; i++) {
+    data.push({
+      air: {
+        temperature: (input.data.readings[i].temperature - 32) * 5/9
+      }
+    });
+  }
+  return {
+    data: data // this is now an array: [ { air: { temperature: ... } }, ... ]
+  }
+}
+```
+
+When the normalizer function returns an array of measurements, Application Server publishes a normalized uplink event for each measurement, while there is still one uplink message event.
+
+## Uplink Example: The Things Node
+
+Here is an example `decodeUplink()` function from The Things Node:
+
+```js
+function decodeUplink(input) {
+  var data = {};
+  var events = {
+    1: "setup",
+    2: "interval",
+    3: "motion",
+    4: "button"
+  };
+  data.event = events[input.fPort];
+  data.battery = (input.bytes[0] << 8) + input.bytes[1];
+  data.light = (input.bytes[2] << 8) + input.bytes[3];
+  data.temperature = (((input.bytes[4] & 0x80 ? input.bytes[4] - 0x100 : input.bytes[4]) << 8) + input.bytes[5]) / 100;
+  var warnings = [];
+  if (data.temperature < -10) {
+    warnings.push("it's cold");
+  }
+  return {
+    data: data,
+    warnings: warnings
+  };
+}
+
+function normalizeUplink(input) {
+  return {
+    data: {
+      air: {
+        temperature: input.data.temperature
+      }
+    }
+  }
+}
+```
+
+Let's assume that the end device observed an event and performed some measurements. The end device transmitted the following binary payload: `0C B2 04 80 F7 AE` (hex encoded) on FPort 4. The Base64 equivalent of that binary payload is `DLIEgPeu` and the JavaScript byte array is `[ 12, 178, 4, 128, 247, 174 ]`. This is all the same, just a different representation.
```

Comment:
Nit: Since `you'd` got corrected to `you would`, maybe `Let's` should be `Let us`.